psl-core/src/main/java/org/linqs/psl/application/util/GroundRules.java

```diff
@@ -193,6 +194,123 @@ public static double getExpectedWeightedCompatibility(WeightedGroundRule groundR
 		return inc * groundRule.getWeight();
 	}
 
+	/** Compute the rounding probability function for a given truth value */
+	public static double roundingProbability(Double truthValue) {
+		return 0.25 + 0.5 * truthValue;
+	}
+
+	/**
+	 * Computes the expected weighted compatibility (1 - incompatibility)
+	 * of a WeightedGroundLogicalRule.
+	 * from independently rounding each {@link RandomVariableAtom} to 1.0 or 0.0
+	 * with probability equal to its current truth value.
+	 *
+	 * WARNING: the result of this function is incorrect if the RandomVariableAtoms
+	 * are subject to any {@link UnweightedGroundRule UnweightedGroundRule}.
+	 *
+	 * WARNING: This method does not account for WeightedGroundLogicalRules that
```

Comment:
Why is this so?
Also, if this is true then we have to find a way around it.
A user will not always know up-front if they will have groundings in a rule (especially with lazy inference).

Reply:
I assumed this limitation still applied, but perhaps not.  At the very least, I wonder if it is wrong for non-lazy inference.